qiskit_nature/second_q/operators/polynomial_tensor.py

```diff
@@ -105,58 +246,97 @@ def is_empty(self) -> bool:
         """Returns whether this tensor is empty or not."""
         return len(self) == 0
 
-    def __getitem__(self, __k: str) -> (np.ndarray | Number):
-        """
-        Returns value matrix in the `PolynomialTensor`.
+    @_optionals.HAS_SPARSE.require_in_call
+    def contains_sparse(self) -> bool:
```

Comment:
Would `issparse` and `isdense` methods make sense that check for all being an instance of SparseArray, or all not being respectively. These might be used internally below too perhaps in todense and tosparse to return self instead of making a copy if it were already all dense matrices or all sparse accordingly?